### PR TITLE
Add a new set of deprecation warnings

### PR DESCRIPTION
**Pull request by hcook**
**Related issue**: 

**Type of change**: bug report | feature request | other enhancement

**Impact**: no functional change | API addition (no impact on existing code) | API modification

**Development Phase**: proposal |  implementation

**Release Notes**

### CODE REVIEW

File: src/main/scala/tilelink/RegisterRouter.scala [hunk 1]
```diff
@@ -138,8 +138,10 @@ class TLRegBundleBase(arg: TLRegBundleArg) extends Bundle
   implicit val p = arg.p
 }
 
+@deprecated("Use HasTLControlRegMap trait in place of TLRegisterRouter+TLRegBundle+TLRegModule", "rocket-chip 1.4")
```
Comment (mwachs5) [inline src/main/scala/tilelink/RegisterRouter.scala]: what does "HasTLControlRegMap" go . with? a LazyModule? LazyModuleImp? Module..?

Comment (mwachs5) [reply src/main/scala/tilelink/RegisterRouter.scala]: also what is a "ControlRegMap"? The name seems a bit specific. Why isn't it just "HasTLRegMap"? Are there other kinds of TL____RegMap?

Comment (hcook) [reply src/main/scala/tilelink/RegisterRouter.scala:141]: if you scroll down in the same file you will see:
```
/** Mix this trait into a RegisterRouter to be able to attach its register map to a TL bus */
trait HasTLControlRegMap { this: RegisterRouter[_] =>
```
and if you check the defintion of `RegisterRouter` you will find it `extends LazyModule`

Comment (hcook) [reply src/main/scala/tilelink/RegisterRouter.scala:141]: A control reg map is presumed to consist of MMIO control registers. This trait adds a node called `controlNode`. It doesn't cover use cases like SPIFlash that have another node for "data"; users can always add extra nodes for additional, non-control maps. Having an BundleBridge IO node is likewise handled by mixing in an additional trait on top of this one.

Comment (hcook) [reply src/main/scala/tilelink/RegisterRouter.scala:141]: `Why isn't it just "HasTLRegMap"? ` ...because what if you have more than one map?

**Pull request closed without merging**